### PR TITLE
Add a CI workflow to build and push Docker image

### PR DESCRIPTION
**Pull request by divergentdave**
This adds a GitHub Actions workflow to build the interop test Docker image, and push the image to Docker Hub (when building from the main branch).

### CODE REVIEW

**Review by jbr — APPROVED** [review]
LGTM
~~does it make sense to build it (but not push) on PRs? Does that happen elsewhere and I missed it?~~

edit: whoops, now I see the check for branch name, neat trick

**Comment by divergentdave** [comment]
Yes, the workflow runs on PRs, but the login and push steps are skipped by a conditional.

**Review by tgeoghegan — APPROVED** [review]
This change accomplishes what it wants, but why use DockerHub when the Janus images go to a Google Artifact Registry? I know that we have been punting on figuring out public access to the Janus GAR, but perhaps we should sort that out instead of re-introducing a dependency on DockerHub?

**Comment by divergentdave** [comment]
I went with this because it's free for us (we don't have to pay for the outgoing network traffic), and it's easy to use as it's the default repository. Technically, we have a read-only dependency on DockerHub already (or a cache of it?) for pulling base images.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -14,14 +14,25 @@ jobs:
...
+      if: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
+      uses: "google-github-actions/auth@v0"
+      with:
+        workload_identity_provider: ${{ secrets.GCP_ARTIFACT_PUBLISHER_WORKLOAD_IDENTITY_PROVIDER }}
+        service_account: ${{ secrets.GCP_ARTIFACT_PUBLISHER_DEPLOY_SERVICE_ACCOUNT }}
+        token_format: "access_token"
+        access_token_lifetime: "3600s"
+        access_token_scopes: "https://www.googleapis.com/auth/cloud-platform"
+        export_environment_variables: true
     - name: Docker Hub Login
```
Comment (tgeoghegan) [inline .github/workflows/docker.yml:29]:
```suggestion
    - name: Docker Login
```

**Review by tgeoghegan — APPROVED** [review]

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**